client/components/data/query-domain-owner-username/index.ts

```diff
@@ -0,0 +1,52 @@
+import { useSelector } from 'react-redux';
+import { useQuerySitePurchases } from 'calypso/components/data/query-site-purchases';
+import useUsersQuery from 'calypso/data/users/use-users-query';
+import { getSitePurchases } from 'calypso/state/purchases/selectors';
+import type { SiteDetails } from '@automattic/data-stores';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+import type { InfiniteData } from 'react-query';
+
+type User = {
+	ID: number;
+	linked_user_ID: number;
+	login: string;
+};
+
+type UsersData = {
+	users: User[];
+};
+
+export function useDomainOwnerUserName(
+	selectedSite: SiteDetails | null | undefined,
+	domain: ResponseDomain | null | undefined
+): string {
+	useQuerySitePurchases( selectedSite?.ID ?? -1 );
+
+	const purchases = useSelector( ( state ) => getSitePurchases( state, selectedSite?.ID ) );
+
+	const selectedSubscriptionId = domain?.subscriptionId ?? '0';
+
+	const domainSubscription = purchases.find(
+		( purchase ) => purchase.id === parseInt( selectedSubscriptionId )
+	);
+
+	const { data, isLoading } = useUsersQuery(
+		selectedSite?.ID,
+		{},
+		{
+			enabled: domainSubscription !== undefined,
+		}
+	);
```

Comment:
```suggestion
	const { data } = useQuery(
		[ 'user', selectedSite?.ID, domainSubscription?.userId ],
		() => wpcom.req.get( `/sites/${ selectedSite?.ID }/users/${ domainSubscription?.userId }` ),
		{ enabled: domainSubscription !== undefined }
	);
```
Not a requirement, but we _could_ query the API directly for a particular user and probably improve loading time. Would this work with the whole `linked_user_ID` vs `ID` question?

Reply:
I tried, but when querying the user when it is a jetpack user id it is not found. it gave me tons of headaches to realize.

I mean, it will work when the site is not a jetpack site, but luckily (or unluckily for my headaches), the domainSubscirption.userId is not the same in the user endpoint.

I've also tried to look for different things, using the same endpoint, and specifying other fields for searching it, with no luck (when the site is a jetpack site, as I said your solution would work flawlessly for non-jetpack sites)

Reply:
Just to make sure I understand: `domainSubscription.userId` can vary between a WordPress.com user ID and a Jetpack external user ID?

Reply:
Nop, the one that vary is the one provided by the User endpoint. Sometimes userId is correct, and sometimes is the Jetpack one.

Reply:
Okay, I see. So looking up `/sites/:site_id/users/:user_id` won't work with `user_id = domainSubscription.userId` for atomic sites, basically?

Reply:
Yes